names/module.py

```diff
@@ -0,0 +1,250 @@
+    @commands.guild_only()
+    @check.acl2(check.ACLevel.MEMBER)
+    @commands.cooldown(rate=5, per=20.0, type=commands.BucketType.user)
+    @nickname_.command(name="set")
+    async def set_nickname(self, ctx, nickname: Optional[str] = None):
+        """Change server nickname"""
+        if not BaseAdminModule.get("boards.karma").enabled:
+            await bot_log.error(
+                ctx.author,
+                ctx.channel,
+                "Repository boards.karma is not loaded. Cannot set nickname.",
+            )
+            await ctx.reply(
+                _(ctx, "Cannot change nickname because mudule karma is not loaded.")
+            )
+            return
+        if nickname is None:
+            await ctx.reply(_(ctx, "Unless you give me nickname I cant't do nothing."))
+            return
+
+        if nickname == ctx.author.display_name:
+            await ctx.reply(
+                _(
+                    ctx,
+                    "New nickname {nickname} is same as current nickname. No need to change.",
+                ).format(nickname=nickname)
+            )
+            return
+
+        price = Price.get("set")
+        if price is None:
+            await bot_log.error(
+                ctx.author,
+                ctx.channel,
+                "Price for command is missing.",
+            )
+            await ctx.reply(
+                _(ctx, "Price for command {command} is missing.").format(command="set")
+            )
+            return
+
+        member = KarmaMember.get_or_add(ctx.author.guild.id, ctx.author.id)
+        if member.value < price.price:
+            await ctx.reply(_(ctx, "No enough karma, try to chat more to earn karma."))
+            return
+
+        for char in ("@", "#", "`", "'", '"'):
+            if char in nickname:
+                await ctx.reply(
+                    _(ctx, "{nickname} include forbidden characters.").format(
+                        nickname=nickname
+                    )
+                )
+                return
+
+        try:
+            await ctx.author.edit(nick=nickname, reason="Nickname purchase")
+        except discord.Forbidden:
+            await ctx.reply(
+                _(ctx, "Cannot change nickname because you have greater privileges.")
+            )
+            return
+
+        member.value += -price.price
+        member.save()
+
+        if ("owo" in nickname.lower()) or ("uwu" in nickname.lower()):
+            await ctx.reply(
+                _(
+                    ctx,
+                    "Congwatuwwation! OwO u wewe suwccessfuwwwy wenamed to {nickname} OwO",
+                ).format(nickname=nickname)
+            )
+        else:
+            await ctx.reply(
+                _(
+                    ctx, "Congratulation! You were successfully renamed to {nickname}."
+                ).format(nickname=nickname)
+            )
+
+    @commands.bot_has_permissions(manage_nicknames=True)
```

Comment:
This permission check can be placed on the parent group, then it will be checked for every subcommand automatically.